x-pack/plugins/osquery/server/routes/pack/update_pack_route.ts

```diff
@@ -34,172 +35,264 @@ import type { PackSavedObject } from '../../common/types';
 import type { PackResponseData } from './types';
 
 export const updatePackRoute = (router: IRouter, osqueryContext: OsqueryAppContext) => {
-  router.put(
-    {
+  router.versioned
+    .put({
+      access: 'public',
       path: '/api/osquery/packs/{id}',
-      validate: {
-        params: schema.object(
-          {
-            id: schema.string(),
-          },
-          { unknowns: 'allow' }
-        ),
-        body: schema.object(
-          {
-            name: schema.maybe(schema.string()),
-            description: schema.maybe(schema.string()),
-            enabled: schema.maybe(schema.boolean()),
-            policy_ids: schema.maybe(schema.arrayOf(schema.string())),
-            shards: schema.maybe(schema.recordOf(schema.string(), schema.number())),
-            queries: schema.maybe(
-              schema.recordOf(
-                schema.string(),
-                schema.object({
-                  query: schema.string(),
-                  interval: schema.maybe(schema.number()),
-                  snapshot: schema.maybe(schema.boolean()),
-                  removed: schema.maybe(schema.boolean()),
-                  platform: schema.maybe(schema.string()),
-                  version: schema.maybe(schema.string()),
-                  ecs_mapping: schema.maybe(
-                    schema.recordOf(
-                      schema.string(),
-                      schema.object({
-                        field: schema.maybe(schema.string()),
-                        value: schema.maybe(
-                          schema.oneOf([schema.string(), schema.arrayOf(schema.string())])
-                        ),
-                      })
-                    )
-                  ),
-                })
-              )
+      options: { tags: [`access:${PLUGIN_ID}-writePacks`] },
+    })
+    .addVersion(
+      {
+        version: API_VERSIONS.public.v1,
+        validate: {
+          request: {
+            params: schema.object(
+              {
+                id: schema.string(),
+              },
+              { unknowns: 'allow' }
+            ),
+            body: schema.object(
```

Comment:
Not required for this pr, but would it be worth having this in a const outside here? It would be more readable I guess,